### PR TITLE
[projmgr] Expose crash when running csolution convert (#1442)

### PR DESCRIPTION
**Pull request by Torbjorn-Svensson**
Contributed by STMicroelectronics

### CODE REVIEW

**Comment by Torbjorn-Svensson** [comment]
@jkrech Here is the reproducer that was asked of me to create in the meeting last week.

File: test/packs/ARM/RteTestGenerator/0.1.0/ARM.RteTestGenerator.pdsc [hunk 1]
```diff
@@ -187,6 +187,12 @@
     <component generator="RteTestExternalGenerator" Cclass="RteTestGenerator" Cgroup="Check Global Generator" Cversion="0.9.0">
       <description>Component addressing external</description>
     </component>
+    <component generator="RteTestGeneratorIdentifier" Cvendor="ARM" Cclass="Device" Cgroup="RteTest Generated Component" Csub="RteTestOverlap" Cversion="1.1.0">
+      <description>Component with overlaping file lists</description>
+      <files>
+        <file category="sourceC" name="Src/RteTestConfig.c" attr="config" version="1.1.0"/>
```
Comment (edriouk) [inline test/packs/ARM/RteTestGenerator/0.1.0/ARM.RteTestGenerator.pdsc:193]: attr="config" is not supported for component files in gpdsc: the files  are alsready in project workspace and should not be copied

**Review by edriouk — CHANGES_REQUESTED** [review]

**Comment by edriouk** [comment]
The tests are failing because a component has been added. This requires updates of golden data in some tests.
In particular, RteModelUtinTests.RteChkTest.Summary require changing number of components from 57 to 58

File: test/packs/ARM/RteTestGenerator/0.1.0/ARM.RteTestGenerator.pdsc (hunk 1 above)
Comment (Torbjorn-Svensson) [reply test/packs/ARM/RteTestGenerator/0.1.0/ARM.RteTestGenerator.pdsc]: While that might be so, this is how I could easily make the crash visible.
If this part is removed, make sure that csolution still crashes on the same statement as without it to ensure that the reported crash is corrected.

**Comment by Torbjorn-Svensson** [comment]
If the .pdsc contains the `attr="config"` on a file that is in a component that is replaced by the gpdsc, then `csolution convert` still crashes in current main (2275f4afe4bf73154dacd5ca4276f925b45e149a), so the crash is still not fixed!

@jkrech I would like to revert the removal of `attr="config"` you did on my branch to re-expose the crash that should be addressed. My PR was about exposing a crash, not "correcting" the wrong usage in the input files. WDYT?

**Comment by jkrech** [comment]
I reverted the inappropriate change to expose the crash.

**Review by edriouk — APPROVED** [review]

**Pull request merged**